pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -1594,4 +1616,9 @@ public String getSegmentName() {
   public void forceCommit() {
     _forceCommitMessageReceived = true;
   }
+
+  @Override
+  public void notifyConsumptionCaughtUp(boolean catchingUpPhase) {
```

Comment:
This call will come in only for the first few consuming segments created during server startup. After that, we will never receive this call.

For example, if a server hosts two partitions, and we start with segment p1_6 and p2_8, then we will get the calls for these two segments after the server has completed its startup.

Now, when p1_6 and p_8 go online, and we create new p1_7 and p2_9 segments, we will never get this call, and the catchup phase will be on forever.

Reply:
This might be the case because the ingestion based status checkers are not aware of the new segments that were not around when the server started. 
Wouldn't that indicate an issue with the ingestion based status checkers itself, then?  Shouldn't it also detect new segments if the segment rolls over during catch-up?

Reply:
The ingestion-based status checkers have a specific purpose -- to answer the question of whether the ingestion has caught up "reasonably" before we can start to field queries after server startup. Once the server is started up, we should be fielding queries no matter whether ingestion is caught up or not. 

The way to handle this situation is to do one of the following:
- Ignore the startup part -- at least for now. Yes, we may show a lag until it catches up, but it is useful metric to observe as well.  I _strongly_ suggest going this approach in the beginning.
- My less preferred approach is to plumb the code so as to keep the ingestion delay trackers disabled until the server has started up completely.  I don't feel comfortable handling it as a part of this PR, since we may introduce unintentional circular dependency.

Reply:
>Another way to handle it is to assume it is caught up in the beginning, and let the status checker to set it as not caught up yet. 

I think that's what my PR (and this one) does.

Reply:
Code changed to use server ready signal now